interfaces/builtin/docker_support.go

```diff
@@ -45,6 +45,16 @@ const dockerSupportBaseDeclarationSlots = `
     deny-auto-connection: true
 `
 
+const dockerSupportConnectedPlugAppArmorCore = `
+# these accesses are necessary for Ubuntu Core 16 and 18, likely due to the version 
+# of apparmor or the kernel which doesn't resolve the upper layer of an 
+# overlayfs mount correctly
+# the accesses show up as runc trying to read from
+# /system-data/var/snap/docker/common/var-lib-docker/overlay2/$SHA/diff/
+/system-data/var/snap/{@{SNAP_NAME},@{SNAP_INSTANCE_NAME}}/common/{,**} rwl,
+/system-data/var/snap/{@{SNAP_NAME},@{SNAP_INSTANCE_NAME}}/@{SNAP_REVISION}/{,**} rwl,
```

Comment:
This is fine since it is in the docker snap's area and we have similar accesses for aufs (without the /system-data bit), but I'm wondering if `.../{,**}` is needed or if `.../{,**/}` is sufficient. I recall when exploring overlays that under some circumstances only the directories were needed, but I don't recall the details OTOH. Again, it is fine as is since these accesses are already allowed via the non-/system-data path.

Reply:
I checked and `.../{,**/}` is sufficient for overlayfs to work. Do you want me to change it to `.../{,**/}`?

Reply:
Done